@3rdweb-sdk/react/hooks/usePayments.ts

```diff
@@ -862,3 +889,181 @@ export function usePaymentsSellerByAccountId(accountId: string) {
+      });
+
+      if (error) {
+        console.error(error);
+      }
+
+      if (data && data.webhook.length > 0) {
+        return data.webhook.map((webhook) => ({
+          id: webhook.id,
+          sellerId: webhook.seller_id,
+          url: webhook.url,
+          isProduction: webhook.is_production,
+          createdAt: new Date(webhook.created_at),
+        })) as PaymentsWebhooksType[];
+      } else {
+        return [] as PaymentsWebhooksType[];
+      }
+    },
+    { enabled: !!paymentsSellerId && !!address },
+  );
+
+  return {
+    ...queryInfo,
+    webhooks: queryInfo.data,
+  };
+}
+
+export const isValidWebhookUrl = (url: string) => {
+  if (!url || !url.startsWith("https://")) {
+    return false;
+  }
+
+  try {
+    const parsedUrl = new URL(url);
+    return !!parsedUrl.href && parsedUrl.protocol === "https:";
+  } catch (e) {
+    return false;
+  }
+};
+
+export type CreateWebhookInput = {
+  url: string;
+  isProduction: boolean;
+};
+
+export function usePaymentsCreateWebhook(accountId: string) {
+  invariant(accountId, "accountId is required");
+  const queryClient = useQueryClient();
+  const address = useAddress();
+  const { paymentsSellerId } = useApiAuthToken();
+
+  const [createWebhookBySellerId] = useInsertWebhookMutation({
+    refetchQueries: [WebhooksBySellerIdDocument],
+  });
+
+  return useMutationWithInvalidate(
+    async (input: CreateWebhookInput) => {
+      invariant(address, "No wallet address found");
+      invariant(paymentsSellerId, "No seller id found");
+      invariant(isValidWebhookUrl(input.url), "Invalid webhook url");
+
+      const webhookInput: Webhook_Insert_Input = {
+        seller_id: paymentsSellerId,
+        url: input.url,
+        is_production: input.isProduction,
+      };
+
+      return createWebhookBySellerId({
+        variables: {
+          object: webhookInput,
+        } as InsertWebhookMutationVariables,
+      });
+    },
+    {
+      onSuccess: () => {
+        return queryClient.invalidateQueries(paymentsKeys.webhooks(accountId));
+      },
+    },
+  );
+}
+
+export type UpdateWebhookInput = {
+  webhookId: string;
+  url?: string;
+  deletedAt?: Date;
+};
+
+export function usePaymentsUpdateWebhook(accountId: string) {
+  invariant(accountId, "accountId is required");
+  const queryClient = useQueryClient();
+  const address = useAddress();
+  const { paymentsSellerId } = useApiAuthToken();
+
+  const [updateWebhookBySellerId] = useUpdateWebhookMutation({
+    refetchQueries: [WebhooksBySellerIdDocument],
+  });
+
+  return useMutationWithInvalidate(
+    async (input: UpdateWebhookInput) => {
+      invariant(address, "No wallet address found");
+      invariant(paymentsSellerId, "No seller id found");
+
+      return updateWebhookBySellerId({
+        variables: {
+          id: input.webhookId,
+          webhookValue: {
+            url: input.url,
+            deleted_at: input.deletedAt,
+          } as Webhook_Set_Input,
```

Comment:
but in this case I don't think you need this? you should get the type safety already